src/Symfony/Component/Console/Style/SymfonyStyle.php

```diff
@@ -437,4 +393,56 @@ private function reduceBuffer($messages)
             return substr($value, -4);
         }, array_merge(array($this->bufferedOutput->fetch()), (array) $messages));
     }
+
+    private function createBlock($messages, $type = null, $style = null, $prefix = ' ', $padding = false, $escape = false)
+    {
+        $indentLength = 0;
+        $lines = array();
+
+        if (null !== $type) {
+            $typePrefix = sprintf('[%s] ', $type);
+            $indentLength = strlen($typePrefix);
+            $lineIndentation = str_repeat(' ', $indentLength);
+        }
+
+        // wrap and add newlines for each element
+        foreach ($messages as $key => $message) {
+            if ($escape) {
+                $message = OutputFormatter::escape($message);
+            }
+
+            $lines = array_merge($lines, explode(PHP_EOL, wordwrap($message, $this->lineLength - Helper::strlen(strip_tags($prefix)) - $indentLength, PHP_EOL, true)));
+
+            // prefix each line with a number of spaces equivalent to the type length
+            if (null !== $type) {
```

Comment:
>7) Symfony\Component\Console\Tests\Style\SymfonyStyleTest::testOutputs with data set #13 ('/Volumes/HD/Sites/perso/contr..._9.php', '/Volumes/HD/Sites/perso/contr..._9.txt')
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
 '
-X [CUSTOM] Custom block
+X          [CUSTOM] ock
 X
 X          Second custom block line

Not all lines need to be processed (the reason of `$lineIndentation === substr($line, 0, $indentLength)`).

ATM this code part (related to the correct alignment/prefixing of blocks) looks good enough to me as it works as expected (get it stable was not so easy, see https://github.com/symfony/symfony/pull/18879). I wouldn't like to break the existing while trying to slightly optimize it "btw" in this PR.

I let you try by yourself, please keep me informed